entity/templates/src/main/java/package/domain/_Entity.java

```diff
@@ -224,6 +224,9 @@ public boolean equals(Object o) {
             return false;
         }
         <%= entityClass %> <%= entityInstance %> = (<%= entityClass %>) o;
+        if(<%= entityInstance %>.id == null || id == null) {
```

Comment:
No, there is no such contract which is needs to be fulfilled.
It needs to be symmetric:
  a.equals(b) == b.equals(a)
Transitive:
  (a.equals(b) && b.equals(c)) => a.equals(c)